docs/architecture/adr-112-proposer-based-timestamps.md

```diff
@@ -9,40 +9,40 @@
  - Feb 2 2022: Synchronize logic for timely with latest version of the spec by @williambanfield
- - Feb 1 2024: Renamed to ADR 112 as basis its adoption ([#1731](https://github.com/cometbft/cometbft/issues/1731)) in CometBFT v1.0 by @cason
+ - Feb 1 2024: Renamed to ADR 112 as basis for its adoption ([#1731](https://github.com/cometbft/cometbft/issues/1731)) in CometBFT v1.0 by @cason
 
 ## Status
 
  **Accepted**
 
 ## Context
 
-CometBFT currently provides a monotonically increasing source of time known as [BFTTime](https://github.com/cometbft/cometbft/blob/main/spec/consensus/bft-time.md).
+CometBFT currently provides a monotonically increasing source of time known as [BFT Time][bfttime].
 This mechanism for producing a source of time is reasonably simple.
-Each correct validator adds a timestamp to each `Precommit` message it sends.
-The timestamp it sends is either the validator's current known Unix time or one millisecond greater than the previous block time, depending on which value is greater.
+Each validator adds a timestamp to each `Precommit` message it sends.
+The timestamp a correct validator sends is either the validator's current known Unix time or one millisecond greater than the previous block time, depending on which value is greater.
 When a block is produced, the proposer chooses the block timestamp as the weighted median of the times in all of the `Precommit` messages the proposer received.
-The weighting is proportional to the amount of voting power, or stake, a validator has on the network.
-This mechanism for producing timestamps is both deterministic and byzantine fault tolerant.
+The weighting is defined by the amount of voting power, or stake, each validator has on the network.
+This mechanism for producing timestamps is both deterministic and Byzantine fault tolerant.
 
 This current mechanism for producing timestamps has a few drawbacks.
 Validators do not have to agree at all on how close the selected block timestamp is to their own currently known Unix time.
-Additionally, any amount of voting power `>1/3` may directly control the block timestamp.
+Additionally, any amount of voting power `>1/3` may control the block timestamp.
 As a result, it is quite possible that the timestamp is not particularly meaningful.
 
-These drawbacks present issues in the CometBFT protocol.
+These drawbacks present issues in CometBFT.
 Timestamps are used by light clients to verify blocks.
-Light clients rely on correspondence between their own currently known Unix time and the block timestamp to verify blocks they see;
-However, their currently known Unix time may be greatly divergent from the block timestamp as a result of the limitations of `BFTTime`.
+Light clients rely on correspondence between their own currently known Unix time and the block timestamp to verify blocks they see.
+However, their currently known Unix time may be greatly divergent from the block timestamp as a result of the limitations of `BFT Time`.
```

Comment:
Should we mention that we want to get rid of timestamps in votes in the long term, so that they are more amenable to aggregation?

Reply:
Yes. There is a commented part referring that vote timestamps become useless. This is commented because we don't change the vote message structure here. But this future/possible approach deserves at least a comment.

Reply:
This is on the future work section. More detailed now, following your suggestion in other comment.